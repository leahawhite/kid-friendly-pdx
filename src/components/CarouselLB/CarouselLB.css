.ril__caption {
  justify-content: center !important;
}

.image-row {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.image-item {
  flex: 100%;
  max-width: 100%;
  position: relative;
  cursor: pointer;
  transition: transform .5s ease;
  margin: 5px;
}

.image-item img {
  display: block;
  width: 100%;
  height: auto;
}

.review-image {
  position: relative;
  width: 150px;
  height: auto;
  cursor: pointer;
  margin: 5px;
 }

.review-image img {
  display: block;
  width: 100%;
  height: 100%;
  border: 1px solid #fff;
  border-radius: 8px;
}

.image-overlay {
  position: absolute; 
  bottom: 0;
  background: rgba(0, 0, 0, 0.5); 
  color: #f1f1f1; 
  width: 100%;
  opacity: 0;
  font-size: 14px;
  line-height: 1.25;
  padding: 10px;
  text-align: center;
}

.review-image:hover .image-overlay {
  transition: .5s ease;
  opacity: 1;
}

