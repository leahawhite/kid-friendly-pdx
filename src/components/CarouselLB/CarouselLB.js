import React, { Component } from 'react';
import Lightbox from 'react-image-lightbox';
import data from '../../data';
import 'react-image-lightbox/style.css'; 
import './CarouselLB.css';
 
export default class LightboxExample extends Component {
  static defaultProps = {
    images: [],
  }
  
  constructor(props) {
    super(props);
 
    this.state = {
      users: data.users,
      photoIndex: 0,
      isOpen: false,
    };
  }

  handleImageClick(index, event) {
    this.setState({ photoIndex: index })
    this.setState({ isOpen: true })
  }
 
  render() {
    const { photoIndex, isOpen } = this.state;
    const { images } = this.props;
    const imageList = images.slice(0, 3).map((image, index) => (
      <div className="image-item" key={index} onClick={this.handleImageClick.bind(this, index)}>
        <img src={image.src} alt={image.title}/>
        <div className="overlay">{image.title}</div>
      </div>
    ), this)
    
    return (
      <div className="image-row">
        {imageList}
 
        {isOpen && (
          <Lightbox
            mainSrc={images[photoIndex].src}
            nextSrc={images[(photoIndex + 1) % images.length].src}
            prevSrc={images[(photoIndex + images.length - 1) % images.length].src}
            imageCaption={images[photoIndex].title}
            onCloseRequest={() => this.setState({ isOpen: false })}
            onMovePrevRequest={() =>
              this.setState({
                photoIndex: (photoIndex + images.length - 1) % images.length,
              })
            }
            onMoveNextRequest={() =>
              this.setState({
                photoIndex: (photoIndex + 1) % images.length,
              })
            }
          />
        )}
      </div>
    );
  }
}