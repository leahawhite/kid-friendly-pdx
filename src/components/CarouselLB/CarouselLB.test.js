import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CarouselLB from './CarouselLB';

describe('CarouselLB component', () => {
  const images = [
    {
      id: 1,
      src: 'https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400',
      title: `Atlas Pizza's 'Jalapeno Popper' pie`,
      place_id: 1,
      user_id: 2,
      review_id: 1,
    },
    {
      id: 2,
      src: 'https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400',
      title: 'A rainbow over Foster Blvd',
      place_id: 1,
      user_id: 1,
      review_id: 2,
    },
  ]
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CarouselLB />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders empty without props', () => {
    const wrapper = shallow(<CarouselLB />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders the images given props', () => {
    const wrapper = shallow(<CarouselLB {...images} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
