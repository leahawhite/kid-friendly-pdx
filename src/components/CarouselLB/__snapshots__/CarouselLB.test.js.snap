// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselLB component renders empty without props 1`] = `
<div
  className="image-row"
/>
`;

exports[`CarouselLB component renders the images given props 1`] = `
<div
  className="image-row"
/>
`;
